src/main/java/org/beanpod/switchboard/controller/DeviceController.java

```diff
@@ -75,24 +76,15 @@ public ResponseEntity createDevice(@RequestBody @Valid DeviceEntity device) {
         return ResponseEntity.ok("Device with serial number " + serialNumber + " Deleted");
     }
 
-    @PutMapping("/{serialNumber}")
+    @PutMapping
     @Transactional
-    public ResponseEntity<String> updateDevice(@PathVariable String serialNumber, @RequestBody DeviceDTO device) {
-
+    public ResponseEntity<DeviceDTO> updateDevice(@RequestBody DeviceDTO device) {
         Optional<DeviceEntity> deviceLookup = service.findDevice(device.getSerialNumber());
         if (deviceLookup.isEmpty()) {
-            throw new ExceptionType.DeviceNotFoundException(serialNumber);
-        }
-
-        if (!serialNumber.equals(device.getSerialNumber())) {
-            throw new ExceptionType.DevicePrimaryKeyRestriction(serialNumber);
-        }
-        int response = service.updateDevice(serialNumber, device);
-
-        if (response != 1) {
-            throw new ExceptionType.DeviceNotUpdated(serialNumber);
+            throw new ExceptionType.DeviceNotFoundException(device.getSerialNumber());
         }
-        return ResponseEntity.ok("Device updated");
+        DeviceEntity deviceEntity = service.save(deviceMapper.toDeviceEntity(device));
```

Comment:
Why did we change to using `save` instead of `updateDevice`?

Reply:
update device was a custome query, save is provided by the jpa interface